golings/cmd/watch.go

```diff
@@ -0,0 +1,148 @@
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/fsnotify/fsnotify"
+	"github.com/mauricioabreu/golings/golings/exercises"
+	"github.com/mauricioabreu/golings/golings/ui"
+	"github.com/spf13/cobra"
+)
+
+func WatchCmd(infoFile string) *cobra.Command {
+	return &cobra.Command{
+		Use:   "watch",
+		Short: "Run a single exercise",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			RunNextExercise(infoFile)
+			reader := bufio.NewReader(os.Stdin)
+			update := make(chan string)
+
+			go WatchEvents(update)
+			go func() {
+				for {
+					select {
+					case f, ok := <-update:
+						if !ok {
+							fmt.Println("OKKKdKKK")
+							return
+						}
+						color.Yellow("=====FILE", f)
+						RunNextExercise(infoFile)
+					}
+				}
+			}()
+
+			for {
+				cmdString, err := reader.ReadString('\n')
+				if err != nil {
+					fmt.Fprintln(os.Stderr, err)
+				}
+
+				cmdStr := strings.TrimSuffix(cmdString, "\n")
+
+				switch cmdStr {
+				case "list":
+					log.Println("List command", cmdString)
+					exs, err := exercises.List(infoFile)
+					if err != nil {
+						color.Red(err.Error())
+						os.Exit(1)
+					}
+					ui.PrintList(os.Stdout, exs)
+
+				case "hint":
+					log.Println("HINT command", cmdString)
+					//@TODO: build a regex to match exercise
+					exs, err := exercises.Find("variables1", infoFile)
+					if err != nil {
+						color.Red(err.Error())
+						os.Exit(1)
+					}
+					color.Yellow(exs.Hint)
+				default:
+					fmt.Errorf("ERROR :/")
+				}
+			}
+		},
+	}
+}
+
+func WatchEvents(updateF chan<- string) {
+	log.Println("===========WATCHER")
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer watcher.Close()
+
+	path, _ := os.Getwd()
+	file_path := fmt.Sprintf("%s/exercises", path)
```

Comment:
This variable should be camelCase, like `filePath` except when you want them to be exported